### PR TITLE
Add remote code fetch

### PR DESCRIPTION
**Pull request by chiting**
Add remote code model to fetch the remote code when order is used with `instant-url` as delivery format

### CODE REVIEW

**Review by kn100 — APPROVED** [review]

File: lib/wegift/models/remote_code.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    def self.get(url)
+      uri = URI(url)
+      conn = Faraday.new(url: "#{uri.scheme}://#{uri.host}")
+
+      response =  conn.get("#{uri.path}?format=json") do |req|
+        req.headers['Content-Type'] = 'application/json'
+      end
+
+      remote_code = self.new(url: url)
+      remote_code = remote_code.parse(response)
```
Comment (kendrikat) [inline lib/wegift/models/remote_code.rb]: why not just `remote_code.parse(response)`

File: lib/wegift/models/remote_code.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    end
+
+    def parse(response)
+      super
+
+      if is_successful?
+        self.class.new(@payload['e_code'])
+      else
+        self
+      end
```
Comment (kendrikat) [inline lib/wegift/models/remote_code.rb:33]: I think you could do this in any case, since it would be initialized with nil, if it's not successfull

**Review by kendrikat — APPROVED** [review]

File: lib/wegift/models/remote_code.rb (hunk 1 above)
Comment (chiting) [reply lib/wegift/models/remote_code.rb]: yeah I don't even know why I did that :/

File: lib/wegift/models/remote_code.rb (hunk 2 above)
Comment (chiting) [reply lib/wegift/models/remote_code.rb:33]: ah yes but I wanted to keep the error attributes set in the parent class' `parse` method

**Pull request merged**